ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts.py

```diff
@@ -0,0 +1,1306 @@
+        node_type_keys.<node_type_key>.vtep
+        """
+        return get(self._node_type_key_data, "vtep", default=False)
+
+    @cached_property
+    def mpls_ler(self):
+        """
+        switch.mpls_ler set based on
+        node_type_keys.<node_type_key>.mpls_ler
+        """
+        return get(self._node_type_key_data, "mpls_ler", default=False)
+
+    @cached_property
+    def ip_addressing(self):
+        """
+        switch.ip_addressing.* set based on
+        templates.ip_addressing.* combined with (overridden by)
+        node_type_keys.<node_type_key>.ip_addressing.*
+        """
+        hostvar_templates = get(self._hostvars, "templates.ip_addressing", default={})
+        node_type_templates = get(self._node_type_key_data, "ip_addressing", default={})
+        if hostvar_templates or node_type_templates:
+            return self._combine(hostvar_templates, node_type_templates, recursive=True, list_merge='replace')
+        else:
+            return {}
+
+    @cached_property
+    def interface_descriptions(self):
+        """
+        switch.interface_descriptions.* set based on
+        templates.interface_descriptions.* combined with (overridden by)
+        node_type_keys.<node_type_key>.interface_descriptions.*
+        """
+        hostvar_templates = get(self._hostvars, "templates.interface_descriptions", default={})
+        node_type_templates = get(self._node_type_key_data, "interface_descriptions", default={})
+        if hostvar_templates or node_type_templates:
+            return self._combine(hostvar_templates, node_type_templates, recursive=True, list_merge='replace')
+        else:
+            return {}
+
+    @cached_property
+    def _switch_data(self):
+        """
+        internal _switch_data containing inherited vars from fabric_topology data model
+
+        Vars are inherited like:
+        <node_type_key>.defaults ->
+            <node_type_key>.node_groups.<group> ->
+                <node_type_key>.node_groups.<group>.nodes.<node> ->
+                <node_type_key>.nodes.<node>
+
+        Returns
+        -------
+        dict
+            node_group : dict
+                Configuration set at the node_group level - including the "nodes" dict.
+                Empty dict if the node is not defined under a node_group.
+            group : str
+                Optional - Name of the matching node_group. Not set if the node is not defined under a node_group.
+            combined : dict
+                Combined configuration after inheritance from all levels
+        """
+        switch_data = {}
+        hostname = self.hostname
+        node_type_config = get(self._hostvars, f"{self.node_type_key}", required=True)
+
+        if hostname in node_type_config.get('nodes', {}):
+            node_config = node_type_config['nodes'][hostname]
+            switch_data['node_group'] = {}
+        else:
+            for node_group in node_type_config.get('node_groups', {}):
```

Comment:
I will not spend too much time on this for now, since we will refactor all this to lists of dicts instead of nested dicts. As this happens, we will have to write some helper functions indeed.